lms/djangoapps/courseware/model_data.py

```diff
@@ -840,6 +840,8 @@ def set_many(self, kv_dict):
             kv_dict (dict): dict mapping from `DjangoKeyValueStore.Key`s to field values
         Raises: DatabaseError if any fields fail to save
         """
+        if self.read_only:
```

Comment:
What is your manual testing strategy?
What if there are certain features that assume a field is present for a user?